diofant/tests/polys/test_rootoftools.py

```diff
@@ -239,6 +239,13 @@ def test_RootOf_conjugate():
     assert r4.evalf() == -r5.evalf()
 
 
+def test_RootOf_as_real_imag():
+    r1, r2 = RootOf(x**3 + x + 3, 1), RootOf(x**3 + x + 3, 2)
+    assert r1.as_real_imag() == (r1/2 + r2/2, -I*(-r2 + r1)/2)
```

Comment:
If you evaluate these expressions numerically do you end up with a small imaginary part?

I think it is better if `expr.as_real_imag()[0].n()` evaluates as a definite real number so I would prefer `re(r1)` here.

Reply:
> If you evaluate these expressions numerically do you end up with a small imaginary part?

Yes, I think so.

> I would prefer re(r1) here.

Hmm, it's possible with ``re(r1 + r2, evaluate=False).evalf()``.  But I doubt this is a good idea to return unevaluated expressions from as_real_imag().

Rather, evalf() should drop imaginary (or real) component if an expression is a real number (or pure imaginary).  Such test should work:
```diff
diff --git a/diofant/core/evalf.py b/diofant/core/evalf.py
index b3e6d39d3b..1747805b19 100644
--- a/diofant/core/evalf.py
+++ b/diofant/core/evalf.py
@@ -1214,7 +1214,7 @@ def evalf(self, dps=15, subs=None, maxn=110, chop=False, strict=True, quad=None)
             re = Float._new(re, p)
         else:
             re = Integer(0)
-        if im:
+        if im and self.conjugate() != self:
             p = max(min(prec, im_acc), 1)
             im = Float._new(im, p)
             return re + im*I
```